controllers/oauth2client_controller.go

```diff
@@ -117,66 +117,68 @@ func (r *OAuth2ClientReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
+				return ctrl.Result{}, registerErr
 			}
-			return ctrl.Result{}, err
+			return ctrl.Result{}, nil
 		}
+		return ctrl.Result{}, err
+	}
 
-		credentials, err := parseSecret(secret)
-		if err != nil {
-			r.Log.Error(err, fmt.Sprintf("secret %s/%s is invalid", secret.Name, secret.Namespace))
-			if updateErr := r.updateReconciliationStatusError(ctx, &oauth2client, hydrav1alpha1.StatusInvalidSecret, err); updateErr != nil {
-				return ctrl.Result{}, updateErr
-			}
-			return ctrl.Result{}, nil
+	credentials, err := parseSecret(secret)
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("secret %s/%s is invalid", secret.Name, secret.Namespace))
+		if updateErr := r.updateReconciliationStatusError(ctx, &oauth2client, hydrav1alpha1.StatusInvalidSecret, err); updateErr != nil {
+			return ctrl.Result{}, updateErr
 		}
+		return ctrl.Result{}, nil
+	}
 
-		hydraClient, err := r.getHydraClientForClient(oauth2client)
-		if err != nil {
-			r.Log.Error(err, fmt.Sprintf(
-				"hydra address %s:%d%s is invalid",
-				oauth2client.Spec.HydraAdmin.URL,
-				oauth2client.Spec.HydraAdmin.Port,
-				oauth2client.Spec.HydraAdmin.Endpoint,
-			))
-			if updateErr := r.updateReconciliationStatusError(ctx, &oauth2client, hydrav1alpha1.StatusInvalidHydraAddress, err); updateErr != nil {
-				return ctrl.Result{}, updateErr
-			}
-			return ctrl.Result{}, nil
+	hydraClient, err := r.getHydraClientForClient(oauth2client)
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf(
+			"hydra address %s:%d%s is invalid",
+			oauth2client.Spec.HydraAdmin.URL,
+			oauth2client.Spec.HydraAdmin.Port,
+			oauth2client.Spec.HydraAdmin.Endpoint,
+		))
+		if updateErr := r.updateReconciliationStatusError(ctx, &oauth2client, hydrav1alpha1.StatusInvalidHydraAddress, err); updateErr != nil {
+			return ctrl.Result{}, updateErr
 		}
+		return ctrl.Result{}, nil
+	}
 
-		fetched, found, err := hydraClient.GetOAuth2Client(string(credentials.ID))
-		if err != nil {
-			return ctrl.Result{}, err
+	fetched, found, err := hydraClient.GetOAuth2Client(string(credentials.ID))
+	if err != nil {
+		return ctrl.Result{}, err
 
-		}
+	}
 
-		if found {
-			if fetched.Owner != fmt.Sprintf("%s/%s", oauth2client.Name, oauth2client.Namespace) {
-				conflictErr := errors.Errorf("ID provided in secret %s/%s is assigned to another resource", secret.Name, secret.Namespace)
-				if updateErr := r.updateReconciliationStatusError(ctx, &oauth2client, hydrav1alpha1.StatusInvalidSecret, conflictErr); updateErr != nil {
-					return ctrl.Result{}, updateErr
-				}
-				return ctrl.Result{}, nil
-			}
+	if found {
+		//conclude reconciliation if the client exists and has not been updated
+		if oauth2client.Generation == oauth2client.Status.ObservedGeneration {
+			return ctrl.Result{}, nil
+		}
 
-			if updateErr := r.updateRegisteredOAuth2Client(ctx, &oauth2client, credentials); updateErr != nil {
+		if fetched.Owner != oauth2client.Name {
```

Comment:
are you sure that it should be changed from `fetched.Owner != fmt.Sprintf("%s/%s", oauth2client.Name, oauth2client.Namespace)` to the current one? This field didn't change: https://github.com/ory/hydra-maester/blob/3422c11749c23918f0d8aa73508f5f515a0c08c2/api/v1alpha1/oauth2client_types.go#L181

Reply:
I have no idea what happened here, I didn't intend to change that :O